### PR TITLE
Add repository snapshots with service, refactor item assigning [SCI-4516]

### PR DESCRIPTION
**Pull request by okriuchykhin**
Jira ticket: [SCI-4516](https://biosistemika.atlassian.net/browse/SCI-4516)

### What was done
Add repository snapshots with service, refactor item assigning

### CODE REVIEW

File: db/migrate/20200331183640_add_repository_snapshots.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class AddRepositorySnapshots < ActiveRecord::Migration[6.0]
+  def up
+    add_column :repositories, :parent_id, :bigint, null: true
+    add_reference :repositories, :my_module
+    add_column :repositories, :type, :string
```
Comment (radiokills) [inline db/migrate/20200331183640_add_repository_snapshots.rb:7]: Is 'type' really the best name for this? Won't it interfere with some reserved keywords?

Comment (okriuchykhin) [reply db/migrate/20200331183640_add_repository_snapshots.rb:7]: It is actually default column name for STI

File: app/models/repository_status_item.rb [hunk 2]
```diff
@@ -1,10 +1,9 @@
 # frozen_string_literal: true
 
 class RepositoryStatusItem < ApplicationRecord
-  validates :repository, :repository_column, :icon, presence: true
+  validates :repository_column, :icon, presence: true
   validates :status, presence: true, length: { minimum: Constants::NAME_MIN_LENGTH,
                                                maximum: Constants::NAME_MAX_LENGTH }
-  belongs_to :repository
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/models/repository_status_item.rb:7]: It should be connected to repository though the column, this relation is not needed here

File: app/models/repository_base.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    repository_columns.pluck(:data_type).each do |data_type|
+      cell_includes << data_type.constantize::PRELOAD_INCLUDE
+    end
+    cell_includes
+  end
+
+  def destroy_discarded(discarded_by_id = nil)
+    self.discarded_by_id = discarded_by_id
+    destroy
+  end
```
[the remark this answers is not in the document]
Comment (okriuchykhin) [reply app/models/repository_base.rb:34]: Removed it from repository

**Pull request merged**